### PR TITLE
Feature/camunda incidents healthcheck

### PR DESCRIPTION
**Pull request by ThomasMinkeRitense**
TP #70093: [BE] Health check for number of Camunda incidents

### CODE REVIEW

File: core/src/main/kotlin/com/ritense/valtimo/camunda/health/IncidentHealthIndicator.kt
Comment (Klaas-Ritense) [inline core/src/main/kotlin/com/ritense/valtimo/camunda/health/IncidentHealthIndicator.kt:1]: Missing copyright header

File: core/src/main/kotlin/com/ritense/valtimo/camunda/health/IncidentHealthIndicator.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import org.camunda.bpm.engine.RuntimeService
+import org.springframework.boot.actuate.health.AbstractHealthIndicator
+import org.springframework.boot.actuate.health.Health
+import org.springframework.boot.actuate.health.Status
+
+class IncidentHealthIndicator (
+    private val runtimeService: RuntimeService
+) : AbstractHealthIndicator() {
+    override fun doHealthCheck(builder: Health.Builder?) {
```
Comment (Klaas-Ritense) [inline core/src/main/kotlin/com/ritense/valtimo/camunda/health/IncidentHealthIndicator.kt:11]: The builder isn't nullable

```suggestion
    override fun doHealthCheck(builder: Health.Builder) {
```

File: core/src/test/kotlin/com/ritense/valtimo/camunda/health/IncidentHealthIndicatorTest.kt
Comment (Klaas-Ritense) [inline core/src/test/kotlin/com/ritense/valtimo/camunda/health/IncidentHealthIndicatorTest.kt:1]: Missing copyright header

File: core/src/main/kotlin/com/ritense/valtimo/actuator/health/ValtimoHealthAggregator.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import org.springframework.boot.actuate.health.Status
+import org.springframework.boot.actuate.health.StatusAggregator
+
+class ValtimoHealthAggregator(): StatusAggregator {
+    override fun getAggregateStatus(statuses: MutableSet<Status>): Status {
+        if(statuses.stream().anyMatch { s -> s.equals(Status.DOWN) }) return Status.DOWN
+        if(statuses.stream().anyMatch { s -> s.equals(Status.OUT_OF_SERVICE) }) return Status.OUT_OF_SERVICE
+        if(statuses.stream().anyMatch { s -> s.equals(Status.UNKNOWN) }) return Status.UNKNOWN
+        return Status.UP
```
Comment (ivo-ritense) [inline core/src/main/kotlin/com/ritense/valtimo/actuator/health/ValtimoHealthAggregator.kt:11]: I would not make UP the default. This means that if we get any unexpected result the service will report status up. I would switch this around and only return UP when all statuses are UP, and make UNKNOWN the default.

**Review by marijnritense — APPROVED** [review]

**Pull request merged**